server/graphql/common/expenses.ts

```diff
@@ -1073,18 +1075,24 @@ export async function createExpense(remoteUser: User | null, expenseData: Expens
   }
 
   // Update payee's location
-  if (!expenseData.payeeLocation?.address && fromCollective.location) {
-    expenseData.payeeLocation = pick(fromCollective.location, ['address', 'country', 'structured']);
+  const existingLocation = await fromCollective.getLocation();
+  if (!(expenseData.payeeLocation?.address || expenseData.payeeLocation?.structured) && existingLocation) {
+    expenseData.payeeLocation = pick(existingLocation, ['address', 'country', 'structured']);
   } else if (
-    expenseData.payeeLocation?.address &&
-    (!fromCollective.location.address || !fromCollective.location.structured)
+    (expenseData.payeeLocation?.address || expenseData.payeeLocation?.structured) &&
+    (!existingLocation.address || !existingLocation.structured)
   ) {
-    // Let's take the opportunity to update collective's location
-    await fromCollective.update({
-      address: expenseData.payeeLocation.address,
-      countryISO: expenseData.payeeLocation.country,
-      data: { ...fromCollective.data, address: expenseData.payeeLocation.structured },
-    });
+    await fromCollective.setLocation(expenseData.payeeLocation);
```

Comment:
Perhaps we should only update Collective locations from payee location for `USER`s, since other types have a public location (in gql v2, v1 slightly more restrictive), might be unexpected to indirectly expose the `payeeLocation` that is private by default.